### DISCUSSION TITLE
Why does a file added to `.gitignore` cause the deployment to fail?

### DISCUSSION THREAD

**Discussion by charnould** [basecamp/kamal#1179] [Q&A] [answered]
I've been using `Kamal` since a while, and it's a great piece of software! Thanks for making it.

However, a question:
- A few seconds ago, my deployments were going fine!
- Whatever the reason is, I've added some files to my `.gitignore` and these files are mandatory to make my app run (I just don't want them in version control)
- When redeploying, it now fails with this error message: `Cannot find module '../assets/config' from ...`, the file I've just added to `.gitignore`...

Why?
Does `Kamal` use code on Github (where `config` isn't)?
Any way to fix this?
Thanks.

**Comment by nickhammond** [comment, marked as answer]
@charnould Kamal by default uses a local git clone, to use the current state of your files you'll need to set `context: "."`.

https://kamal-deploy.org/docs/configuration/builders/#build-context

**Reply by charnould** [reply]
Thanks!
